### PR TITLE
#145839283 : Fixed unit tests  in apiserver and servicebroker

### PR DESCRIPTION
**Pull request by rohitsharma04**

### CODE REVIEW

File: api/test/unit/config/settings_test.js [hunk 1]
```diff
@@ -90,24 +95,28 @@ describe('config setting Test Suite', function() {
         it('Should return false', function() {
           settings.db.maxConnections = null;
           expect(settings.validate().valid).to.equal(false);
+          expect(settings.validate().message).to.equal("maxConnections is required");
```
Comment (ghaih) [inline api/test/unit/config/settings_test.js]: In the test for port , when port is null,the message is "The port must be a number". Here when maxConnections  is null, message is "maxConnections is required". Can have similar messages at both spots.

File: api/test/unit/config/settings_test.js [hunk 2]
```diff
@@ -90,24 +95,28 @@ describe('config setting Test Suite', function() {
...
           delete settings.db.maxConnections
           expect(settings.validate().valid).to.equal(false);
+          expect(settings.validate().message).to.equal("maxConnections is required");
         })
       });
       context('When db.maxConnections is not an integer', function() {
         it('Should return false', function() {
           settings.db.maxConnections = "10";
           expect(settings.validate().valid).to.equal(false);
+          expect(settings.validate().message).to.equal("maxConnections is required");
```
Comment (ghaih) [inline api/test/unit/config/settings_test.js]: Should we say maxConnections  should be a number.

File: api/test/unit/config/settings_test.js (hunk 1 above)
Comment (rohitsharma04) [inline api/test/unit/config/settings_test.js]: done.

File: api/lib/config/setting.js [hunk 3]
```diff
@@ -41,28 +39,40 @@ module.exports = function(settingsObj) {
...
       return {valid:false,message:"minConnections is required"};
     }
+    if(typeof(settings.db.minConnections) != "number") {
+      return {valid:false,message:"minConnections must be a number"};
+    }
     if(settings.db.minConnections < 0) {
       return {valid:false,message:"minConnections must be greater than 0"};
     }
-    if(typeof(settings.db.idleTimeout) != "number") {
+    if(!settings.db.idleTimeout || settings.db.idleTimeout == null){
```
Comment (ghaih) [inline api/lib/config/setting.js]: Little different from the way done for the rest with typeof?
 Also can make a generic function to do the required check  and another one to check if number?

Comment (rohitsharma04) [inline api/lib/config/setting.js]: @ghaih updated the code for both apiserver and servicebroker.

**Comment by rohitsharma04** [comment]
@ghaih and @kanekoh any more comments ?

**Comment by paltanmoy** [comment]
LGTM

File: api/lib/config/setting.js [hunk 4]
```diff
@@ -40,71 +38,119 @@ module.exports = function(settingsObj) {
...
   settings.validate = function(){
-    if(typeof(settings.port) != "number") {
-      return {valid:false,message:"The port must be a number"};
+    if (isMissing(settings.port)){
+      return{valid:false, message:"port is required"}
+    }
+    if (!isNumber(settings.port)) {
+      return {valid:false,message:"port must be a number"};
     }
     if (settings.port < 0 || settings.port > 65536) {
```
Comment (rohitsharma04) [inline api/lib/config/setting.js]: @kanekoh  done.

**Pull request merged**